src/extra/sprite.cpp

```diff
@@ -0,0 +1,453 @@
+#include "./sprite.hpp"
+
+namespace chainblocks {
+namespace Sprite {
+
+struct Sheet {
+  static inline Type ObjType{
+      {CBType::Object, {.object = {.vendorId = CoreCC, .typeId = SheetCC}}}};
+  static inline Type VarType = Type::VariableOf(ObjType);
+
+  static inline ObjectVar<Sheet> Var{"Sprite-Sheet", CoreCC, SheetCC};
+
+  static CBTypesInfo inputTypes() { return CoreInfo::NoneType; }
+
+  static CBTypesInfo outputTypes() { return Sheet::ObjType; }
+
+  static CBParametersInfo parameters() { return _params; }
+
+  void setParam(int index, const CBVar &value) {
+    switch (index) {
+    case 0:
+      _atlas = value;
+      break;
+
+    default:
+      throw CBException("Parameter out of range.");
+    }
+  }
+
+  CBVar getParam(int index) {
+    switch (index) {
+    case 0:
+      return _atlas;
+    }
+    throw CBException("Parameter out of range.");
+  }
+
+  void cleanup() { _atlas.cleanup(); }
+
+  void warmup(CBContext *context) { _atlas.warmup(context); }
+
+  CBVar activate(CBContext *context, const CBVar &input) {
+    auto table = _atlas.get().payload.tableValue;
+    auto api = table.api;
+
+    auto size = api->tableAt(table, "size")->payload.seqValue;
+    this->width = size.elements[0].payload.intValue;
+    this->height = size.elements[1].payload.intValue;
+
+    this->premultiply = api->tableAt(table, "pma")->payload.boolValue;
+
+    auto regions = api->tableAt(table, "regions")->payload.seqValue;
+    this->regions.reserve(regions.len);
+    for (uint32_t i = 0; i < regions.len; ++i) {
+      auto table = regions.elements[i].payload.tableValue;
+      auto api = table.api;
+
+      auto region = Region{};
+
+      auto bounds = api->tableAt(table, "bounds")->payload.seqValue;
+      region.width = bounds.elements[2].payload.intValue;
+      region.height = bounds.elements[3].payload.intValue;
+
+      if (api->tableContains(table, "uvs")) {

```

Comment:
I'm thinking about if we actually need uvs. maybe we can just always specify pixel coordinates in the skeleton instead.

We could support both but it causes some confusion if both are specified like in the test sprite sheets since it leaves some values unused.

Reply:
How does it work in the shader? Doesn't it need to have texture coordinates?

Reply:
I meant leave them out of the skeleton file and just always compute them using the bounds

Reply:
It was to save a few computations 😉 but maybe not worth it.